treeherder/webapp/api/push.py

```diff
@@ -70,6 +73,11 @@ def list(self, request, project):
                 )
 
             pushes = pushes.filter(repository=repository)
+    
+        if search_query:
+            pushes = pushes.annotate(
+                search_vector=SearchVector('commits__revision', 'commits__author', 'commits__comments')
```

Comment:
You'll need to use the exact same SearchVector configuration as the one in the index, so that the index is used.